### PR TITLE
MAINT: Apply deprecations

### PR DESCRIPTION
**Pull request by larsoner**
Post-release deprecations

Should be squash+merged with a `[circle deploy]` so we see if everything works and 1.5.dev docs get deployed

### CODE REVIEW

**Comment by larsoner** [comment]
... and switching back again. I don't think we need to wait for the installers to finish. The filenames are predictable so might as well update the links now (done) and their build will probably finish before CircleCI manages to deploy the docs 4h from now :)

File: mne/time_frequency/spectrum.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ def plot(
         %(picks_good_data_noref)s
```
Comment (drammock) [inline mne/time_frequency/spectrum.py:585]:
```suggestion
        %(picks_all_data_noref)s
```

**Review by drammock — COMMENTED** [review]
there should be some `TODOs` in `mne/report/report.py` too, related to the spectrum picks/exclude

**Comment by larsoner** [comment]
Why no auto merge? Seems to be working okay

**Pull request merged**